### PR TITLE
stylo: Improve pseudo element restyling and support the content property.

### PR DESCRIPTION
**Pull request by emilio**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because geckolib.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12895)

### CODE REVIEW

**Comment by emilio** [comment]
Bugzilla bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1292618

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -1355,6 +1355,88 @@ fn static_assert() {
...
+            T::normal => {
+                if !self.gecko.mContents.is_empty() {
+                    unsafe {
+                        Gecko_ClearStyleContents(&mut self.gecko);
+                    }
+                }
+            },
+            T::Content(items) => {
+                // Ensure destructors run, otherwise we could leak.
+                if !self.gecko.mContents.is_empty() {
```
Comment (bholley) [inline components/style/properties/gecko.mako.rs]: Given that we do this in both branches, can we just pull it out and do it unconditionally before?

**Comment by bholley** [comment]
r=me with those changes, really sorry for the long delay

@bors-servo delegate+

**Comment by emilio** [comment]
@bors-servo: r=bholley p=1

 * blocking stylo's Gecko-side changes

**Pull request merged**